components/BottomNavigation/src/MDCBottomNavigationBarController.m

```diff
@@ -148,6 +164,38 @@ - (BOOL)bottomNavigationBar:(MDCBottomNavigationBar *)bottomNavigationBar
   return YES;
 }
 
+#pragma mark - Key Value Observation Methods
+
+- (void)observeValueForKeyPath:(NSString *)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary<NSKeyValueChangeKey, id> *)change
+                       context:(void *)context {
+  if (context != kObservationContext) {
+    [super observeValueForKeyPath:keyPath ofObject:object change:change context:context];
+    return;
+  }
+
+  id newValue = [change objectForKey:NSKeyValueChangeNewKey];
+  if (object == self.navigationBar &&
+      [keyPath isEqualToString:NSStringFromSelector(@selector(items))] &&
+      [newValue isKindOfClass:[NSArray class]]) {
+    [self didUpdateNavigationBarItemsWithNewValue:(NSArray *)newValue];
+  }
+}
+
+- (void)didUpdateNavigationBarItemsWithNewValue:(NSArray *)items {
+  // Verify tab bar items correspond with the view controllers tab bar items.
+  NSAssert(items.count == self.viewControllers.count,
```

Comment:
Please make this a conditional (`if`) check and log + return early. If this fails in production, the NSAssert will not fire and the loop below is likely to access an invalid array index.

Reply:
Done!